node/src/database/db_migrations.rs

```diff
@@ -1291,6 +1423,72 @@ mod tests {
         assert!(assertion.is_some()) //means there is a table named 'just_garden' now
     }
 
+    #[test]
+    fn execute_upon_transaction_handles_also_error_from_stm_with_params() {
+        let dir_path = ensure_node_home_directory_exists(
+            "db_migrations",
+            "execute_upon_transaction_handles_also_error_from_stm_with_params",
+        );
+        let db_path = dir_path.join("test_database.db");
+        let conn = Connection::open(&db_path).unwrap();
+        conn.execute(
+            "CREATE TABLE botanic_garden (
+                        name TEXT,
+                        count integer
+                    )",
+            [],
+        )
+        .unwrap();
+        let statement_1_simple =
+            "INSERT INTO botanic_garden (name,count) VALUES ('sun_flowers', 100)";
+        let statement_2_good = StatementWithRusqliteParams {
+            sql_stm: "select * from botanic_garden".to_string(),
+            params: RefCell::new(Some({
+                let params: &[&dyn ToSql] = [].as_slice();
+                params
+            })),
+        };
+        let statement_3_bad = StatementWithRusqliteParams {
+            sql_stm: "select * from foo".to_string(),
+            params: RefCell::new(Some(&[&"another_whatever"])),
+        };
+        //we won't get down here, error from the execution of statement_3 immediately terminate the circuit
+        let statement_4_demonstrative = StatementWithRusqliteParams {
+            sql_stm: "select * from bar".to_string(),
+            params: RefCell::new(Some(&[&"also_whatever"])),
+        };
+        let set_of_sql_statements: &[&dyn StatementObject] = &[
+            &statement_1_simple,
+            &statement_2_good,
+            &statement_3_bad,
+            &statement_4_demonstrative,
+        ];
+        let mut conn_wrapper = ConnectionWrapperReal::new(conn);
+        let config = DBMigratorInnerConfiguration::new();
+        let external_params = make_external_migration_parameters();
+        let subject = DBMigrationUtilitiesReal::new(&mut conn_wrapper, config).unwrap();
+
+        let result = subject
+            .make_mig_declaration_utils(&external_params, &Logger::new("test logger"))
+            .execute_upon_transaction(set_of_sql_statements);
+
+        match result {
+            //I was unable recreate the first field in the returned error
+            Err(Error::SqliteFailure(_, err_msg_opt)) => {
+                assert_eq!(err_msg_opt, Some("no such table: foo".to_string()))
+            }
+            x => panic!("we expected SqliteFailure(..) but got: {:?}", x),
+        }
+        let assert_conn = Connection::open(&db_path).unwrap();
```

Comment:
Please add a comment here about how the reason the query comes back empty is because the error caused the transaction to be aborted.